app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -380,6 +386,23 @@ class BrowserActivity : DuckDuckGoActivity(), BookmarkDialogCreationListener {
         webView.restoreState(bundle)
     }
 
+    override fun onCreateContextMenu(menu: ContextMenu, view: View, menuInfo: ContextMenu.ContextMenuInfo?) {
+        webView.hitTestResult?.let {
+            viewModel.userLongPressedInWebView(it, menu)
+        }
+    }
+
+    override fun onContextItemSelected(item: MenuItem): Boolean {
+        webView.hitTestResult?.let {
+            var url = it.extra
+            if(viewModel.userSelectedItemFromLongPressMenu(url, item)) {
```

Comment:
Clean up formatting

Reply:
What site/image are you seeing that's giving you `downloadfile.bin`? I'm seeing sensible filenames by default when I'm testing.

<img width="416" alt="screen shot 2018-02-07 at 09 41 57" src="https://user-images.githubusercontent.com/1336281/35909337-2ca9b0e8-0beb-11e8-9a5c-c06f5212f875.png">

Reply:
I had tried a few images on espc.com and the guardian.co.uk and they are all `downloadfile.bin` in our app but fine in Chrome. I just tried bbc and that looks good in our app too

Reply:
Thanks; will investigate